pkg/runtime/grpc/instances.go

```diff
@@ -118,7 +118,16 @@ func (r *Runtime) createGadgetInstance(gadgetCtx runtime.GadgetContext, runtimeP
 	var err error
 	instanceID := runtimeParams.Get(ParamID).AsString()
 	instanceName := runtimeParams.Get(ParamName).AsString()
-
+	// checks if the instance name already exists
+	instances, err := r.GetGadgetInstances(gadgetCtx.Context(), runtimeParams)
+	if err != nil {
+		return fmt.Errorf("getting gadget instances: %w", err)
+	}
+	for _, instance := range instances {
+		if instance.Name == instanceName {
+			return fmt.Errorf("gadget instance with name %q already exists", instanceName)
+		}
+	}
```

Comment:
I think it should be implemented server side on https://github.com/inspektor-gadget/inspektor-gadget/blob/7eafb8e0b244bfe6fecf1bfffb139e447bf3a4a5/pkg/gadget-service/store/k8s-configmap-store/configmapstore.go#L234

Look how it's done for the filestore implementation: 

https://github.com/inspektor-gadget/inspektor-gadget/blob/06f618de976fb77420ac70dde0d07d1a2996cd70/pkg/gadget-service/store/file-store/filestore.go#L136-L143

Reply:
Done! @mauriciovasquezbernal, let me know your thoughts on the changes now.